contracts/ito.sol

```diff
@@ -62,49 +72,35 @@ contract HappyTokenPool {
 
     // claim success event
     event ClaimSuccess (
-        bytes32 id,
-        address claimer,
+        bytes32 indexed id,
+        address indexed claimer,
         uint256 timestamp,
         uint256 to_value,
         address token_address
     );
 
     // swap pool destruct success event
     event DestructSuccess (
-        bytes32 id,
-        address token_address,
+        bytes32 indexed id,
+        address indexed token_address,
         uint256 remaining_balance,
         uint128[] exchanged_values
     );
 
     // single token withdrawl from a swap pool success even
     event WithdrawSuccess (
-        bytes32 id,
-        address token_address,
+        bytes32 indexed id,
+        address indexed token_address,
         uint256 withdraw_balance
     );
 
-    modifier creatorOnly {
-        require(msg.sender == contract_creator, "Contract Creator Only");
-        _;
-    }
-
     using SafeERC20 for IERC20;
-    uint32 nonce;
-    uint224 base_time;                 // timestamp = base_time + delta to save gas
-    address public contract_creator;
+    uint32 constant base_time = 1616976000; // timestamp = base_time + delta to save gas
```

Comment:
to generalize this contract, please set these parameters in the constructor instead of setting them as a static number.

Reply:
added